catalogd/internal/serverutil/serverutil.go

```diff
@@ -97,3 +93,12 @@ func logrLoggingHandler(l logr.Logger, handler http.Handler) http.Handler {
 		l.Info("handled request", "host", host, "username", username, "method", params.Request.Method, "uri", uri, "protocol", params.Request.Proto, "status", params.StatusCode, "size", params.Size)
 	})
 }
+
+func storageServerHandlerWrapped(l logr.Logger, cfg CatalogServerConfig) http.Handler {
```

Comment:
Nit (follow-up): We could probably change `CatalogServerConfig` to have an `http.Handler` field instead of the entire `LocalStorage` field?